src/cluster.jl

```diff
@@ -1185,7 +1276,7 @@ function deregister_worker(pg, pid)
             # Notify the cluster manager of this workers death
             manage(w.manager, w.id, w.config, :deregister)
             if PGRP.topology !== :all_to_all || isclusterlazy()
-                for rpid in workers()
+                for rpid in filter(!=(myid()), workers())
```

Comment:
No that only removes `pid` (the ID of the dead worker) from the internal list. `deregister_worker()` is called on the live workers when a worker `pid` goes down, so when we recurse our own ID will still be in `workers()`. This didn't cause issues before because the function can handle being called recursively, but now it would cause the `exited` callbacks to be called twice.